### PR TITLE
document ipns pubsub

### LINKED ISSUE #4902: Document IPNS over pubsub experiment
It appears that we're missing documentation for this feature in docs/experimental-features.md.

### PR DESCRIPTION
**Pull request by vyzo**
Closes #4902

### CODE REVIEW

**Comment by vyzo** [comment]
@magik6k yes, this is true; i'll add a note.

File: docs/experimental-features.md [hunk 1]
```diff
@@ -365,3 +365,30 @@ ipfs config --json Experimental.ShardingEnabled true
...
+
+### In Version
+
+0.4.13
+
+### State
+
+Experimental, default-disabled.
+
+Utilizes pubsub for publishing ipns records in real time.
```
Comment (Mr0grog) [inline docs/experimental-features.md]: Can we say slightly more about how it works/what it does? e.g:

> When you publish an IPNS record, a message about the record will also be published via pubsub (note: does it get repeatedly sent [and roughly what frequency?] or just once at publish time?). When you resolve an IPNS address, your daemon will subscribe to the pubsub topic for that name so it is immediately informed of changes.

I’m not even totally sure my understanding there is correct, so please fix if it’s not!

File: docs/experimental-features.md [hunk 2]
```diff
@@ -365,3 +365,30 @@ ipfs config --json Experimental.ShardingEnabled true
...
 - [ ] Make sure that objects that don't have to be sharded aren't
 - [ ] Generalize sharding and define a new layer between IPLD and IPFS
+
+---
+
+## IPNS pubsub
+
+### In Version
+
+0.4.13
```
Comment (Stebalien) [inline docs/experimental-features.md]: Was it in this version? I thought this only made it into 0.4.14.

File: docs/experimental-features.md (hunk 1 above)
Comment (Stebalien) [reply docs/experimental-features.md]: Really, the important part is: what should the user expect and why would they want to try this feature out?

Comment (Mr0grog) [reply docs/experimental-features.md]: Yes. Exactly that ^

File: docs/experimental-features.md (hunk 2 above)
Comment (vyzo) [reply docs/experimental-features.md]: uhm ok, not sure either.

File: docs/experimental-features.md (hunk 1 above)
Comment (vyzo) [reply docs/experimental-features.md]: ok fine, I'll add _another_ sentence :)

File: docs/experimental-features.md [hunk 3]
```diff
@@ -365,3 +365,38 @@ ipfs config --json Experimental.ShardingEnabled true
...
+### State
+
+Experimental, default-disabled.
+
+Utilizes pubsub for publishing ipns records in real time.
+
+When it is enabled:
+- IPNS publishers push records to a name-specific pubsub topic,
+  in addition to publishing to the DHT.
+- IPNS resolvers subscribe to the name-specific topic and receive newly
```
Comment (Stebalien) [inline docs/experimental-features.md]: At the moment, it's a bit unclear what this actually means, I'd go with something like:

> IPNS resolvers subscribe to name-specific topics on first resolutions to speedup future resolutions of the same name.

Comment (vyzo) [reply docs/experimental-features.md]: i'll massage it a bit.

File: docs/experimental-features.md [hunk 4]
```diff
@@ -383,9 +383,12 @@ Utilizes pubsub for publishing ipns records in real time.
 When it is enabled:
 - IPNS publishers push records to a name-specific pubsub topic,
   in addition to publishing to the DHT.
-- IPNS resolvers subscribe to the name-specific topic and receive newly
-  published records through pubsub in real time. Note that the initial resolution
-  still goes through the DHT, as there is no message history in pubsub.
+- IPNS resolvers subscribe to the name-specific topic on first
+  resoltion and receive subsequently published records through pubsub
```
Comment (Mr0grog) [inline docs/experimental-features.md]: Typo on “resol**u**tion”

Comment (vyzo) [reply docs/experimental-features.md]: fixed, thanks.

**Pull request merged**